### PR TITLE
Tasks Due Function

### LINKED ISSUE #122: Find Applications with Outstanding Tasks
As an anxious user, I can quickly finish outstanding tasks

### PR DESCRIPTION
**Pull request by HM33-Stu**
Implement basic structure for tasks due function
Resolves #122

### CODE REVIEW

File: src/main/java/seedu/address/logic/InternshipLogicManager.java [hunk 1]
```diff
@@ -62,6 +66,25 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         return commandResult;
     }
 
+    public List<Task> getDueTasks() {
+        ObservableList<Task> allTasks = model.getInternshipData().getDueTaskList();
```
Comment (LWS49) [inline src/main/java/seedu/address/logic/InternshipLogicManager.java:70]: how do you get the DueTaskList? currently they all return null

Comment (HM33-Stu) [reply src/main/java/seedu/address/logic/InternshipLogicManager.java:70]: Its created in the InternshipData.java
Hope I didnt misunderstand the qn

File: src/main/java/seedu/address/logic/InternshipLogicManager.java [hunk 2]
```diff
@@ -62,6 +66,25 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
...
+                        return ""; // Replace null with an empty string
+                    }
+                    return task.getDeadline().toString(); // Convert deadline to string for comparison
+                }))
+                .collect(Collectors.toList());
+
+        // Return top 3 tasks or fewer if the total number of tasks is less than 3
+        return sortedTasks.stream()
+                .limit(3)
+                .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (LWS49) [reply src/main/java/seedu/address/logic/InternshipLogicManager.java:85]: I suggest that you make use of the functions used in internshipFindCommand to find the internship entries that have the specific task. i will edit this comment for further detail, not sure how to implement yet but just a suggestion.
the alternative would be to pass the internship alongside the task into this and have the internship and the task paired together throughout, which probably would be much harder to do.

**Comment by HM33-Stu** [comment]
Team has decided to not proceed with this functionality

**Pull request closed without merging**